<?xml version="1.0" ?>
<launch>
    <arg name="robot_ip" default="$(env FRANKA_IP)"/>
    <arg name="object" default="shelf"/>
    <arg name="object_reference_position" default="1.57" if="$(eval arg('object') == 'shelf')"/>

    <include file="$(find mppi_sliding)/launch/load.launch">
        <arg name="debug" value="false"/>
        <arg name="object" value="$(arg object)"/>
    </include>

    <include file="$(find franka_gripper)/launch/franka_gripper.launch">
        <arg name="robot_ip" value="$(arg robot_ip)" />
    </include>

    <rosparam command="load" file="$(find mppi_sliding)/config/experiments/controllers_mug.yaml" />
    <rosparam command="load" file="$(find mppi_sliding)/config/experiments/franka_control_node.yaml" />
    <rosparam command="load" file="$(find mppi_sliding)/params/params.yaml"/>

    <node name="franka_control" pkg="franka_control" type="franka_control_node" output="screen" required="true">
        <param name="robot_ip" value="$(arg robot_ip)" />
        <param name="object_reference_position" value="$(arg object_reference_position)"/>
        <param name="activation_file" value="/home/boyang/.raisim/activation.raisim"/>
        <param name="references_file" value="$(find mppi_sliding)/config/experiments/references.yaml"/>
        <param name="experiment_name" value="$(arg experiment_name)"/>
    </node>

    <node name="state_controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="ManipulationController"/>


</launch>